rust/hyperlane-core/src/db/hyperlane_db.rs

```diff
@@ -195,53 +197,112 @@ impl HyperlaneDB {
             return Ok(false);
         }
         // Set the gas payment as processed
-        self.store_gas_payment_meta_processed(meta)?;
+        self.store_tx_meta_processed(meta)?;
 
         // Update the total gas payment for the message to include the payment
-        self.update_gas_payment_for_message_id(&gas_payment_with_meta.payment)?;
+        self.update_gas_payment_for_message_id(gas_payment_with_meta.payment)?;
 
         // Return true to indicate the gas payment was processed for the first time
         Ok(true)
     }
 
-    /// Record a gas payment, identified by its metadata, as processed
-    fn store_gas_payment_meta_processed(
+    /// If the provided gas expenditure, identified by its metadata, has not
+    /// been processed, processes the gas expenditure and records it as
+    /// processed. Returns whether the gas expenditure was processed for the
+    /// first time.
+    pub fn process_gas_expenditure(
         &self,
-        gas_payment_meta: &InterchainGasPaymentMeta,
-    ) -> Result<()> {
-        self.store_keyed_encodable(GAS_PAYMENT_META_PROCESSED, gas_payment_meta, &true)
+        gas_expenditure_with_meta: &GasExpenditureWithMeta,
+    ) -> Result<bool> {
+        let meta = &gas_expenditure_with_meta.meta;
+        // If the gas payment has already been processed, do nothing
+        if self.retrieve_tx_meta_processed(meta)? {
```

Comment:
Ig in my head it just made sense to track what transactions were actually processed since it feels really concerning to juts update an accumulator without any record of how it got to that value. It does add complexity and I don't think there is any real value to keeping track of the transactions at this time so I am going to remove it. We should have other means by which to find this info anyway if we want to debug.